Add Peter's first examples.

Peter230655 opened the pull request:
3D n body pendulum, where the bodies can rotate around the massless rods connecting one body to the next one. If they hit each other, they are ideally elastic and slick. Visualisation using pythreejs.

Please edit below based on the type of PR. It is then the duty of reviewers to check off the items as they are completed. If any questions are not relevant to your particular pull request, a reviewer will simply check it off as done.

Example
-------

   - [ ] There are no merge conflicts with the `master` branch.
   - [x] Examples rst file is placed in `docs/examples/`.
   - [ ] An entry with a 200x200 pixel image has been added to
     `docs/index.rst`.
   - [ ] RsT and image filenames are kebab-case (lower case hyphen-separated).
   - [x] jupyter_sphinx RsT directives are used to ensure example is executed
     on build.
   - [x] jupyter_sphinx `ipynb` and `py` download links are included.
   - [ ] The example has adequate text and figures (preferably SVG) explaining
     the problem.
   - [ ] All reviewer comments have been addressed.

moorepants commented on Pendulum_with_elastic_balls.rst:
```diff
@@ -0,0 +1,540 @@
+
+
+
+.. image:: output_3_2.png
```
Delete these image directives. They will be generated automatically.

moorepants commented on Pendulum_with_elastic_balls.rst:
```diff
@@ -0,0 +1,540 @@
+
+.. parsed-literal::
+
+    deviation of total energy from being constant is 0.47471 % of max. total energy
```
Remove any outputs like this, as they will be generated automatically.

moorepants commented on Pendulum_with_elastic_balls.rst:
```diff
@@ -0,0 +1,540 @@
+.. code:: ipython3
```
```suggestion
.. jupyter-execute::
```

moorepants commented on Pendulum_with_elastic_balls.rst:
```diff
@@ -0,0 +1,540 @@
+    import pythreejs as p3js
+
+
+.. jupiter-execute::
```
```suggestion
.. jupyter-execute::
```

moorepants replied:
Make sure to spell "jupyter" correctly everywhere.

moorepants commented on Pendulum_with_elastic_balls.rst:
```diff
@@ -0,0 +1,540 @@
+    - of course I do not know for sure.
+    
+    
+    cse = True reduced the time it take to integrate substantially, by a factor of over 50
```
Move any test like this outside of the code cells (outside of `.. jupyter-execute::`.

moorepants commented on Pendulum_with_elastic_balls.rst:
```diff
@@ -0,0 +1,540 @@
+    Dmc_loc_lam = sm.lambdify(qL + pL, Dmc_loc, cse=cse1)
+    punkt_loc_lam = sm.lambdify(qL + pL, punkt_loc, cse=cse1)
+    
+    print('it took {:.3f} sec to set up Kanes equations'.format(time.time() - start))
```
This is one big code cell. If you want to display any intermediate outputs in this code cell, break it up into more code cells with more `.. jupyter-execute::` lines.

moorepants commented on Pendulum_with_elastic_balls.rst:
```diff
@@ -0,0 +1,540 @@
+    force free symbols {m1, g, iXX, r, l, k, iYY, iZZ, m, t}
+    force contains 12757 operations 
+    
+    it took 7.903 sec to set up Kanes equations
```
Remove any outputs.

moorepants commented:
You'll need to upload these files to the correct directory, which is: https://github.com/pydy/pydy/tree/master/docs/examples

Also, review how those files are written and make sure to include the same structure for example the opening likes look like:

```rst
==============================
Carvallo-Whipple Bicycle Model
==============================

.. note::

   You can download this example as a Python script:
   :jupyter-download:script:`carvallo-whipple` or Jupyter notebook:
   :jupyter-download:notebook:`carvallo-whipple`.
```

moorepants commented:
Congrats on making your first pull request! There are a few things to fix before it will load properly in the documentation website, but you are very close!

Peter230655 commented:
I changed something here, but not sure I really understood what I should do.
Peter

On Sun 2. Oct 2022 at 09:47 Jason K. Moore ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In Pendulum_with_elastic_balls.rst
> <https://github.com/pydy/pydy/pull/467#discussion_r985195593>:
>
> > +
> +    '''
> +    A pendulum consisting on n massless rods of length l. A ball of mass m and radius r is attached to each
> +    rod, such that the rod goes through the center of each ball. the center of the ball is fixed at the middle
> +    of the rod. the ball rotates around the rod.
> +    If two balls collide, they are ideally elastic, with 'spring constant' k.
> +    The collision is modeled using the sm.Heaviside(..) function.
> +    The balls are ideally slick, so collisions will not affect their rotation.
> +
> +    The total energy does not always remain constant.
> +    As method = 'Radau' in solve_ivp gives much 'better' results than no method (e.g. 15% deviation vs. 0.3% in
> +    some cases), I assume that this is due to numerical errors in the integration.
> +    - of course I do not know for sure.
> +
> +
> +    cse = True reduced the time it take to integrate substantially, by a factor of over 50
>
> Move any test like this outside of the code cells (outside of ..
> jupyter-execute::.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pydy/pydy/pull/467#pullrequestreview-1127604332>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUUVWGZ7DT5ZRCD2IB3WBE4YXANCNFSM6AAAAAAQ2OGATI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

Peter230655 commented:
1.
I thought I loaded my file to this directory, into a fork, like I
understood you?

2.

*.. note::

   You can download this example as a Python script:
   :jupyter-download:script:`3D, n body pendulum` or Jupyter notebook:
   :jupyter-download:notebook:`3D, n body pendulum`.*

Should I add this at the top of my RST file?

Peter

On Sun 2. Oct 2022 at 09:50 Jason K. Moore ***@***.***> wrote:

> You'll need to upload these files to the correct directory, which is:
> https://github.com/pydy/pydy/tree/master/docs/examples
>
> Also, review how those files are written and make sure to include the same
> structure for example the opening likes look like:
>
> ==============================
> Carvallo-Whipple Bicycle Model==============================
>
> .. note::
>
>    You can download this example as a Python script:
>    :jupyter-download:script:`carvallo-whipple` or Jupyter notebook:
>    :jupyter-download:notebook:`carvallo-whipple`.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pydy/pydy/pull/467#issuecomment-1264577372>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUXZI37QLRMKXLMQTLDWBE5FPANCNFSM6AAAAAAQ2OGATI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

Peter230655 commented:
Thanks!
I corrected, what I understood, had two questions.

On Sun 2. Oct 2022 at 09:51 Jason K. Moore ***@***.***> wrote:

> Congrats on making your first pull request! There are a few things to fix
> before it will load properly in the documentation website, but you are very
> close!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pydy/pydy/pull/467#issuecomment-1264577522>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUQU32XOJREVMARGDCTWBE5IHANCNFSM6AAAAAAQ2OGATI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

Peter230655 commented:
It is long, but I am unsure, which intermediate results may be of interest.
Most of my examples look like this.
If you could give me a hint, I could change in future samples.
Thanks! Peter

On Sun 2. Oct 2022 at 09:48 Jason K. Moore ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In Pendulum_with_elastic_balls.rst
> <https://github.com/pydy/pydy/pull/467#discussion_r985195752>:
>
> > +
> +    # Lambdification
> +    qL = q1 + u1
> +    pL = [m, m1, g, r, l, iXX, iYY, iZZ, reibung, k]
> +
> +    MM_lam = sm.lambdify(qL + pL, MM, cse=cse1)
> +    force_lam = sm.lambdify(qL + pL, force, cse=cse1)
> +
> +    pot_lam = sm.lambdify(qL + pL, pot_energie, cse=cse1)
> +    kin_lam = sm.lambdify(qL + pL, kin_energie, cse=cse1)
> +    spring_lam = sm.lambdify(qL + pL, spring_energie, cse=cse1)
> +
> +    Dmc_loc_lam = sm.lambdify(qL + pL, Dmc_loc, cse=cse1)
> +    punkt_loc_lam = sm.lambdify(qL + pL, punkt_loc, cse=cse1)
> +
> +    print('it took {:.3f} sec to set up Kanes equations'.format(time.time() - start))
>
> This is one big code cell. If you want to display any intermediate outputs
> in this code cell, break it up into more code cells with more ..
> jupyter-execute:: lines.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pydy/pydy/pull/467#pullrequestreview-1127604442>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUUY7C6VXU5RX4WY2ITWBE44HANCNFSM6AAAAAAQ2OGATI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

moorepants commented:
> I thought I loaded my file to this directory, into a fork, like I understood you?

If you click the "Files Changed" tab on the pull request you see that the `.rst` files are in the top level pydy directory, not in the docs examples directory.

moorepants commented:
> *.. note::
> 
>    You can download this example as a Python script:
>    :jupyter-download:script:`3D, n body pendulum` or Jupyter notebook:
>    :jupyter-downloadnotebook`3D, n body pendulum`.*
> 
> Should I add this at the top of my RST file?

Yes, but please name the files using lower case and hyphens, like the files already in `docs/examples`, e.g. `my-pydy-example.rst`.

moorepants commented:
> It is long, but I am unsure, which intermediate results may be of interest.
> Most of my examples look like this.
> If you could give me a hint, I could change in future samples.

I don't have any preference. If you don't want/need to show anything intermediate, then don't.

moorepants commented:
One of your files says it is 44 thousand lines long. Is that correct?

Peter230655 commented:
My original file was short, maybe 200 lines.
When I looked at the RST file it contained pages full of symbols, I guess
those 44,000 lines. No idea what they mean.
Maybe I best remove it and try again.

On Sun 2. Oct 2022 at 12:55 Jason K. Moore ***@***.***> wrote:

> One of your files says it is 44 thousand lines long. Is that correct?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pydy/pydy/pull/467#issuecomment-1264612680>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUVYQIH6GUP2MGKXVILWBFS2LANCNFSM6AAAAAAQ2OGATI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

Peter230655 commented:
I was too stupid to find this!
I found pull request tabs, which I clicked, but never saw a a „Flies
Changed“ tab.
Sorry, Peter

On Sun 2. Oct 2022 at 12:50 Jason K. Moore ***@***.***> wrote:

> I thought I loaded my file to this directory, into a fork, like I
> understood you?
>
> If you click the "Files Changed" tab on the pull request you see that the
> .rst files are in the top level pydy directory, not in the docs examples
> directory.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pydy/pydy/pull/467#issuecomment-1264611631>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUXFCGHPA3FEZ5FP3E3WBFSFRANCNFSM6AAAAAAQ2OGATI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

moorepants commented:
See files changed here:
![image](https://user-images.githubusercontent.com/276007/193451727-865d7198-8330-4aaa-ac66-dc1ad955b139.png)

moorepants commented:
> My original file was short, maybe 200 lines.
> When I looked at the RST file it contained pages full of symbols, I guess
> those 44,000 lines. No idea what they mean.
> Maybe I best remove it and try again.
> 

One tip is to "clear all output" in Jupyter before converting to rst. We don't need the outputs in these files, as they will get generating automatically later.

Peter230655 commented:
I will do this.
This ‚monster output seems to be caused by this HTML command I need on the
iPad to show matplotlib animations.
When I ‚delete it‘ by putting # before this line, the RST file becomes
normal.

Thanks, Peter

On Sun 2. Oct 2022 at 13:39 Jason K. Moore ***@***.***> wrote:

> My original file was short, maybe 200 lines.
> When I looked at the RST file it contained pages full of symbols, I guess
> those 44,000 lines. No idea what they mean.
> Maybe I best remove it and try again.
>
> One tip is to "clear all output" in Jupyter before converting to rst. We
> don't need the outputs in these files, as they will get generating
> automatically later.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pydy/pydy/pull/467#issuecomment-1264622531>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUWDLVBQ7FI4JT3DSQTWBFYARANCNFSM6AAAAAAQ2OGATI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

Peter230655 commented:
On Sun 2. Oct 2022 at 14:39 Peter Stahlecker ***@***.***>
wrote:

> I will do this.
> This ‚monster output seems to be caused by this HTML command I need on the
> iPad to show matplotlib animations.
> When I ‚delete it‘ by putting # before this line, the RST file becomes
> normal.
>
> Thanks, Peter
>
> On Sun 2. Oct 2022 at 13:39 Jason K. Moore ***@***.***>
> wrote:
>
>> My original file was short, maybe 200 lines.
>> When I looked at the RST file it contained pages full of symbols, I guess
>> those 44,000 lines. No idea what they mean.
>> Maybe I best remove it and try again.
>>
>> One tip is to "clear all output" in Jupyter before converting to rst. We
>> don't need the outputs in these files, as they will get generating
>> automatically later.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/pydy/pydy/pull/467#issuecomment-1264622531>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AT5MQUWDLVBQ7FI4JT3DSQTWBFYARANCNFSM6AAAAAAQ2OGATI>
>> .
>> You are receiving this because you modified the open/close state.Message
>> ID: ***@***.***>
>>
> --
> Best regards,
>
> Peter Stahlecker
>
-- 
Best regards,

Peter Stahlecker

Peter230655 commented:
Dear Jason,

Maybe you will change your mind about me not being trouble! :-)

Please see the screen shot below.
This is ‚my fork‘, as I understand it.

If I click ‚pendulum with elastic balls‘ (blue in the picture), the version
which I corrected (changed jupiter into jupyter..) comes up.
If I click the ‚Update Pendulum… (to the right of it) the *un*corrected
version comes up.
Is this the way it should be, or did I do something wrong?

Thanks, Peter




On Sun 2. Oct 2022 at 15:04 Peter Stahlecker ***@***.***>
wrote:

>
>
> On Sun 2. Oct 2022 at 14:39 Peter Stahlecker ***@***.***>
> wrote:
>
>> I will do this.
>> This ‚monster output seems to be caused by this HTML command I need on
>> the iPad to show matplotlib animations.
>> When I ‚delete it‘ by putting # before this line, the RST file becomes
>> normal.
>>
>> Thanks, Peter
>>
>> On Sun 2. Oct 2022 at 13:39 Jason K. Moore ***@***.***>
>> wrote:
>>
>>> My original file was short, maybe 200 lines.
>>> When I looked at the RST file it contained pages full of symbols, I guess
>>> those 44,000 lines. No idea what they mean.
>>> Maybe I best remove it and try again.
>>>
>>> One tip is to "clear all output" in Jupyter before converting to rst. We
>>> don't need the outputs in these files, as they will get generating
>>> automatically later.
>>>
>>> —
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/pydy/pydy/pull/467#issuecomment-1264622531>, or
>>> unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/AT5MQUWDLVBQ7FI4JT3DSQTWBFYARANCNFSM6AAAAAAQ2OGATI>
>>> .
>>> You are receiving this because you modified the open/close state.Message
>>> ID: ***@***.***>
>>>
>> --
>> Best regards,
>>
>> Peter Stahlecker
>>
> --
> Best regards,
>
> Peter Stahlecker
>
-- 
Best regards,

Peter Stahlecker

moorepants commented:
A bunch of the errors seem to be related to cse results changing in a new sympy version (don't worry about them). I'll have to fix in another PR.

```
======================================================================
FAIL: pydy.codegen.tests.test_c_code.TestCMatrixGenerator.test_doprint
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/share/miniconda/envs/test-environment/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/runner/work/pydy/pydy/pydy/codegen/tests/test_c_code.py", line 1001, in test_doprint
    assert source == '\n'.join(lines[:1] + lines[2:])
AssertionError

======================================================================
FAIL: pydy.codegen.tests.test_c_code.TestCMatrixGenerator.test_generate_code_blocks
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/share/miniconda/envs/test-environment/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/runner/work/pydy/pydy/pydy/codegen/tests/test_c_code.py", line 553, in test_generate_code_blocks
    assert v == expected[k]
AssertionError

======================================================================
FAIL: pydy.codegen.tests.test_c_code.TestCMatrixGenerator.test_generate_cse
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/share/miniconda/envs/test-environment/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/runner/work/pydy/pydy/pydy/codegen/tests/test_c_code.py", line 177, in test_generate_cse
    assert self.generator.subexprs == expected_subexprs
AssertionError

======================================================================
FAIL: pydy.codegen.tests.test_octave_code.test_OctaveMatrixGenerator
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/share/miniconda/envs/test-environment/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/runner/work/pydy/pydy/pydy/codegen/tests/test_octave_code.py", line 145, in test_OctaveMatrixGenerator
    assert g.doprint() == expected_m_file_1p2
AssertionError
```

Peter230655 commented:
Dear Jason,

Two questions, I have:

1.
Anything I can do right now, or do I wait until this „cse issue“ is taken
care of?

2.
Should I upload another example (without cse in lambdify(..), or better
wait until the issue above is solved?

Thanks & take care!

Peter

On Sun 2. Oct 2022 at 16:07 Peter Stahlecker ***@***.***>
wrote:

> Makes me feel a bit better. :-)
> Should I change to cse = False in my sm.lambdify(..) ?
> But cse = True speeds up numerical integration so dramatically, I found
> factors up to over 100 fold.
> Peter
>
> On Sun 2. Oct 2022 at 16:01 Jason K. Moore ***@***.***>
> wrote:
>
>> A bunch of the errors seem to be related to cse results changing in a new
>> sympy version (don't worry about them). I'll have to fix in another PR.
>>
>> ======================================================================
>> FAIL: pydy.codegen.tests.test_c_code.TestCMatrixGenerator.test_doprint
>> ----------------------------------------------------------------------
>> Traceback (most recent call last):
>>   File "/usr/share/miniconda/envs/test-environment/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
>>     self.test(*self.arg)
>>   File "/home/runner/work/pydy/pydy/pydy/codegen/tests/test_c_code.py", line 1001, in test_doprint
>>     assert source == '\n'.join(lines[:1] + lines[2:])
>> AssertionError
>>
>> ======================================================================
>> FAIL: pydy.codegen.tests.test_c_code.TestCMatrixGenerator.test_generate_code_blocks
>> ----------------------------------------------------------------------
>> Traceback (most recent call last):
>>   File "/usr/share/miniconda/envs/test-environment/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
>>     self.test(*self.arg)
>>   File "/home/runner/work/pydy/pydy/pydy/codegen/tests/test_c_code.py", line 553, in test_generate_code_blocks
>>     assert v == expected[k]
>> AssertionError
>>
>> ======================================================================
>> FAIL: pydy.codegen.tests.test_c_code.TestCMatrixGenerator.test_generate_cse
>> ----------------------------------------------------------------------
>> Traceback (most recent call last):
>>   File "/usr/share/miniconda/envs/test-environment/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
>>     self.test(*self.arg)
>>   File "/home/runner/work/pydy/pydy/pydy/codegen/tests/test_c_code.py", line 177, in test_generate_cse
>>     assert self.generator.subexprs == expected_subexprs
>> AssertionError
>>
>> ======================================================================
>> FAIL: pydy.codegen.tests.test_octave_code.test_OctaveMatrixGenerator
>> ----------------------------------------------------------------------
>> Traceback (most recent call last):
>>   File "/usr/share/miniconda/envs/test-environment/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
>>     self.test(*self.arg)
>>   File "/home/runner/work/pydy/pydy/pydy/codegen/tests/test_octave_code.py", line 145, in test_OctaveMatrixGenerator
>>     assert g.doprint() == expected_m_file_1p2
>> AssertionError
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/pydy/pydy/pull/467#issuecomment-1264651019>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AT5MQUS3RHMVD4H3EV7WYOTWBGIUNANCNFSM6AAAAAAQ2OGATI>
>> .
>> You are receiving this because you modified the open/close state.Message
>> ID: ***@***.***>
>>
> --
> Best regards,
>
> Peter Stahlecker
>
-- 
Best regards,

Peter Stahlecker

moorepants commented:
I'm trying to fix the general pydy tests here: https://github.com/pydy/pydy/pull/468

moorepants commented:
This file has to be edited: https://github.com/pydy/pydy/blob/master/docs/index.rst

moorepants commented:
This is the issue: https://github.com/pydy/pydy/issues/465 the pydy tests fail with sympy 1.10 and 1.11. If you make this contribution work with SymPy 1.9, that's the path of least resistance to get this merged.

Peter230655 commented:
I deleted this monster file, I managed to do it! :-)
I think, I will upload the next one only after this first one is acceptable.
This shouldd save you some work, I hope.

I corrected the jupyter spellings, I added the note, and I changed the name.
Should the file name the the same as the name in the note?

Thanks! Peter

On Sun 2. Oct 2022 at 12:59 Peter Stahlecker ***@***.***>
wrote:

> My original file was short, maybe 200 lines.
> When I looked at the RST file it contained pages full of symbols, I guess
> those 44,000 lines. No idea what they mean.
> Maybe I best remove it and try again.
>
> On Sun 2. Oct 2022 at 12:55 Jason K. Moore ***@***.***>
> wrote:
>
>> One of your files says it is 44 thousand lines long. Is that correct?
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/pydy/pydy/pull/467#issuecomment-1264612680>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AT5MQUVYQIH6GUP2MGKXVILWBFS2LANCNFSM6AAAAAAQ2OGATI>
>> .
>> You are receiving this because you modified the open/close state.Message
>> ID: ***@***.***>
>>
> --
> Best regards,
>
> Peter Stahlecker
>
-- 
Best regards,

Peter Stahlecker

Peter230655 commented:
Makes me feel a bit better. :-)
Should I change to cse = False in my sm.lambdify(..) ?
But cse = True speeds up numerical integration so dramatically, I found
factors up to over 100 fold.
Peter

On Sun 2. Oct 2022 at 16:01 Jason K. Moore ***@***.***> wrote:

> A bunch of the errors seem to be related to cse results changing in a new
> sympy version (don't worry about them). I'll have to fix in another PR.
>
> ======================================================================
> FAIL: pydy.codegen.tests.test_c_code.TestCMatrixGenerator.test_doprint
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/usr/share/miniconda/envs/test-environment/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
>     self.test(*self.arg)
>   File "/home/runner/work/pydy/pydy/pydy/codegen/tests/test_c_code.py", line 1001, in test_doprint
>     assert source == '\n'.join(lines[:1] + lines[2:])
> AssertionError
>
> ======================================================================
> FAIL: pydy.codegen.tests.test_c_code.TestCMatrixGenerator.test_generate_code_blocks
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/usr/share/miniconda/envs/test-environment/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
>     self.test(*self.arg)
>   File "/home/runner/work/pydy/pydy/pydy/codegen/tests/test_c_code.py", line 553, in test_generate_code_blocks
>     assert v == expected[k]
> AssertionError
>
> ======================================================================
> FAIL: pydy.codegen.tests.test_c_code.TestCMatrixGenerator.test_generate_cse
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/usr/share/miniconda/envs/test-environment/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
>     self.test(*self.arg)
>   File "/home/runner/work/pydy/pydy/pydy/codegen/tests/test_c_code.py", line 177, in test_generate_cse
>     assert self.generator.subexprs == expected_subexprs
> AssertionError
>
> ======================================================================
> FAIL: pydy.codegen.tests.test_octave_code.test_OctaveMatrixGenerator
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/usr/share/miniconda/envs/test-environment/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
>     self.test(*self.arg)
>   File "/home/runner/work/pydy/pydy/pydy/codegen/tests/test_octave_code.py", line 145, in test_OctaveMatrixGenerator
>     assert g.doprint() == expected_m_file_1p2
> AssertionError
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pydy/pydy/pull/467#issuecomment-1264651019>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUS3RHMVD4H3EV7WYOTWBGIUNANCNFSM6AAAAAAQ2OGATI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

Peter230655 commented:
I took this route to modify my file (take out ‚cse‘ in lambdify(..).
I hope this was correct.

On Tue 11. Oct 2022 at 10:51 Jason K. Moore ***@***.***>
wrote:

> This file has to be edited:
> https://github.com/pydy/pydy/blob/master/docs/index.rst
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pydy/pydy/pull/467#issuecomment-1274337662>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUV236GYO5FBHLC24GTWCUTB5ANCNFSM6AAAAAAQ2OGATI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

Peter230655 commented:
*1.*

*You also haven't updated the front page of the docs to have links to your
new examples.*
Sorry, but I have no idea what this means / what I should do.

2.

*Your files are still not in the docs/examples/ directory. That has to be
fixed*.
I have no idea, what to do here, sorry!

3.
I think, I should change the name of my example to

*3d_n_body_pendulum,*

But again no idea how to do it.


On Tue 11. Oct 2022 at 10:50 Jason K. Moore ***@***.***>
wrote:

> Anything I can do right now, or do I wait until this „cse issue“ is taken
> care of?
>
> Your files are still not in the docs/examples/ directory. That has to be
> fixed.
>
> You also haven't updated the front page of the docs to have links to your
> new examples.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pydy/pydy/pull/467#issuecomment-1274335694>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUUHPXOJCPQ6VFDYGVTWCUS5HANCNFSM6AAAAAAQ2OGATI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

moorepants commented:
> *You also haven't updated the front page of the docs to have links to your
> new examples.*
> Sorry, but I have no idea what this means / what I should do.

You have to edit this file:

https://github.com/pydy/pydy/blob/master/docs/index.rst

moorepants commented:
> *Your files are still not in the docs/examples/ directory. That has to be
> fixed*.
> I have no idea, what to do here, sorry!

The "files changed" tab shows:
![image](https://user-images.githubusercontent.com/276007/195105874-65e78d8d-9d82-411f-ae24-5acf3d199074.png)

Note that `Pendulum_with_elastic_balls.rst` should be `docs/examples/Pendulum_with_elastic_balls.rst`

Peter230655 commented:
I think, I finally managed.

On Tue 11. Oct 2022 at 15:38 Jason K. Moore ***@***.***>
wrote:

> This is how you rename a file:
> https://docs.github.com/en/repositories/working-with-files/managing-files/renaming-a-file
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pydy/pydy/pull/467#issuecomment-1274700447>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUTJ3HZY4VR7NOMVKNTWCVUT5ANCNFSM6AAAAAAQ2OGATI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

moorepants commented on docs/examples/3d_n_body_pendulum.rst:
```diff
@@ -0,0 +1,490 @@
+    some cases), I assume that this is due to numerical errors in the integration.
+    - of course I do not know for sure.
+    '''
+    start = time.time()
```
Move text to left and remove jupyter execute. Remove the quotations `'''`.

moorepants commented:
Also this might be helpful: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html it gives you the syntax for `.rst` documents.

moorepants commented:
I think you may have it!

Let's see if the build works.

Peter230655 commented:
I think, it failed. No idea, why.  :-((

On Wed 12. Oct 2022 at 19:06 Jason K. Moore ***@***.***>
wrote:

> I think you may have it!
>
> Let's see if the build works.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pydy/pydy/pull/467#issuecomment-1276485376>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQURO5WLU7D4GEX6CLD3WC3V25ANCNFSM6AAAAAAQ2OGATI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

moorepants commented on docs/examples/3d_n_body_pendulum.rst:
```diff
@@ -0,0 +1,492 @@
+   :jupyter-download:notebook:`3d_n_body_pendulum`.
+
+.. jupyter-execute::
+    import sympy as sm
```
The error I'm seeing is this:
```
/home/runner/work/pydy/pydy/docs/examples/3d_n_body_pendulum.rst.rst:11: WARNING: Ignoring inline code in Jupyter cell included from "examples/import sympy as sm
```

I think that you need a blank line after every `.. jupyter-execute::` line. One blank line before the code starts.

Peter230655 commented:
Done, I hope!

On Wed 12. Oct 2022 at 20:07 Jason K. Moore ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In docs/examples/3d_n_body_pendulum.rst
> <https://github.com/pydy/pydy/pull/467#discussion_r993761666>:
>
> > @@ -0,0 +1,492 @@
> +
> +===================
> +3d_n_body_pendulum
> +===================
> +.. note::
> +
> +   You can download this example as a Python script:
> +   :jupyter-download:script:`3d_n_body_pendulum` or Jupyter notebook:
> +   :jupyter-download:notebook:`3d_n_body_pendulum`.
> +
> +.. jupyter-execute::
> +    import sympy as sm
>
> The error I'm seeing is this:
>
> /home/runner/work/pydy/pydy/docs/examples/3d_n_body_pendulum.rst.rst:11: WARNING: Ignoring inline code in Jupyter cell included from "examples/import sympy as sm
>
> I think that you need a blank line after every .. jupyter-execute:: line.
> One blank line before the code starts.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pydy/pydy/pull/467#pullrequestreview-1139523431>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUQYCG6Q4GGOVZ4KQGDWC3475ANCNFSM6AAAAAAQ2OGATI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

moorepants commented:
Current error:

```
Extension error:
Cell raised uncaught exception:
  File <tokenize>:37
    return np.array(sol).T[0]
    ^
IndentationError: unindent does not match any outer indentation level
```

click the red x docs/ Build ... details link to see these.

Peter230655 commented:
Corrected, I hope!

On Wed 12. Oct 2022 at 20:44, Jason K. Moore ***@***.***>
wrote:

> Current error:
>
> Extension error:
> Cell raised uncaught exception:
>   File <tokenize>:37
>     return np.array(sol).T[0]
>     ^
> IndentationError: unindent does not match any outer indentation level
>
> click the red x docs/ Build ... details link to see these.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pydy/pydy/pull/467#issuecomment-1276590310>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUVQV67LBDB5V7AUJ3DWC4BI3ANCNFSM6AAAAAAQ2OGATI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

moorepants commented:
It worked!

The only other thing is to add a link to it on the index.rst file. Otherwise no one will find the page.

Peter230655 commented:
Good, thanks a LOT for your help!!
How do I add this link?

On Wed 12. Oct 2022 at 21:22, Jason K. Moore ***@***.***>
wrote:

> It worked!
>
> The only other thing is to add a link to it on the index.rst file.
> Otherwise no one will find the page.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pydy/pydy/pull/467#issuecomment-1276636659>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUSBMCGMSNIDC6RXUILWC4FYNANCNFSM6AAAAAAQ2OGATI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

moorepants commented:
You have to edit this file: https://github.com/pydy/pydy/blob/master/docs/index.rst

Peter230655 commented:
I tried, I hope I did it right…

On Thu 13. Oct 2022 at 06:58 Jason K. Moore ***@***.***>
wrote:

> You have to edit this file:
> https://github.com/pydy/pydy/blob/master/docs/index.rst
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pydy/pydy/pull/467#issuecomment-1277032814>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUTTAR5J6OOJXCXXZTTWC6JI3ANCNFSM6AAAAAAQ2OGATI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 
Best regards,

Peter Stahlecker

Peter230655 commented:
I did it right??? :-))

On Thu 13. Oct 2022 at 07:53, Jason K. Moore ***@***.***>
wrote:

> Merged #467 <https://github.com/pydy/pydy/pull/467> into master.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pydy/pydy/pull/467#event-7577643349>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUTTHWNYCLIYA7IMVC3WC6PVHANCNFSM6AAAAAAQ2OGATI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 
Best regards,

Peter Stahlecker
